docs/source/features/parallelisms.rst

```diff
@@ -112,6 +130,16 @@ NeMo integrates Tensor Parallelism through the implementation from Megatron Core
 
 For detailed API usage and additional configurations, consult the `Megatron Core Developer Guide <https://docs.nvidia.com/Megatron-Core/developer-guide/latest/api-guide/tensor_parallel.html>`_.
 
+FSDP + Tensor Parallelism
+~~~~~~~~~
+
+NeMo supports FSDP along with tensor parallelism. This is done by restricting the model state sharding to the data-parallel domain.
+Using FSDP with tensor parallelism can be helpful when the model doesn't have sufficient parallelism to deploy on a large scale training system with the data-parallel mapping; e.g., running a model with the global batch size of 1024 on 2048 GPUs.
+Also, tensor parallelism enables FSDP feasibility by reducing the model state size and the activation size per GPU, thus lower the FSDP communication overhead and the activation memory overhead.
+
+FSDP + TP works by enabling FSDP (``fsdp=true``) and setting ``tensor_model_parllel_size > 1``.
+The user should unset ``CUDA_DEVICE_MAX_CONNECTIONS`` environment variable to enable that sets the number of GPU kernel queue to overlap of the FSDP communication with computation kernels.
```

Comment:
To enable FSDP to work with TP, set FSDP (fsdp=true) and tensor_model_parallel_size greater than 1. To enable FSDP communication overlap with computation kernels, unset the CUDA_DEVICE_MAX_CONNECTIONS environment variable. This adjustment affects the number of GPU kernel queues.